app/main.py

```diff
@@ -1,3 +1,65 @@
+from __future__ import annotations
+from typing import Any
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Any) -> Distance:
+        if not isinstance(other, Distance):
+            return Distance(self.km + other)
+        else:
+            return Distance(self.km + other.km)
+
+    def __iadd__(self, other: Any) -> Distance:
+        if not isinstance(other, Distance):
+            self.km += other
+            return self
+        else:
+            self.km += other.km
+            return self
+
+    def __mul__(self, other: Any) -> Distance:
+        if not isinstance(other, Distance):
+            return Distance(self.km * other)
```

Comment:
The `__mul__` method does not handle the case where `other` is a `Distance` instance. Consider adding logic to handle or raise an exception for unsupported operations.